returnn/datasets/postprocessing.py

```diff
@@ -228,10 +252,47 @@ def _make_tensor_template_from_input(self, data_key: str) -> Tensor:
         return Tensor(data_key, dims=dims, dtype=dtype, sparse_dim=sparse_dim)
 
 
+T = TypeVar("T", TensorDict, Iterator[TensorDict])
+
+
+def compose(*postprocessing_funcs: Callable):
+    """
+    Composes multiple postprocessing functions into one by sequential application,
+    i.e. compose(f, g)(x) = (f ∘ g)(x) = f(g(x)).
+
+    Can either compose ``map_seq``-style single-segment processor functions or ``map_seq_stream``-style
+    iterators operating on multiple segments.
+
+    The functions are applied in reverse order, i.e. last argument first.
+
+    :return: composite function applying :param:``postprocessing_funcs`` in reverse order.
+    """
+
+    def wrapper(arg: T, **kwargs) -> T:
+        """composite postprocessing function"""
+
+        # If we are passed an iterator do not check for the concrete type (which may be some generator),
+        # but for parent `Iterator` type instead.
+        arg_type = Iterator if isinstance(arg, Iterator) else type(arg)
+
+        for i, func in enumerate(reversed(postprocessing_funcs)):
```

Comment:
Don't do it reversed. That just adds confusion.
```suggestion
        for i, func in enumerate(postprocessing_funcs):
```

Reply:
But this is how its done in math, and how I'd expect it to work.

Reply:
What do you mean by "in math"? You mean when you write "f ∘ g"? But why does that matter?

For some algorithm/code, you would always assume the most simple logic, which is not reversing it but just using the original order. Only because some specific notation in math has the reverse order is irrelevant for this? Despite "f ∘ g", there are other notations being used which have the original order, not reversed order (see [here](https://en.wikipedia.org/wiki/Function_composition#Alternative_notations)).

And also, all other similar functions in common frameworks also don't reverse the order but keep the original order, e.g. [`torch.nn.Sequential`](https://pytorch.org/docs/stable/generated/torch.nn.Sequential.html).

But maybe instead of calling it `compose`, call it `Sequential`, to make that relation even more obvious.

Reply:
I would also find the "reversed" order more natural (because of writing "f ∘ g", but also the order how one would write it in code `foo(bar(baz(x))) =: compose(foo, bar, baz)(x)` ). 
So I would call it "reversed order" now and "in order" before.

I do not care much which order is implemented, though,  as long as it is documented well.

Reply:
I think with the rename to `Sequential`, this ambiguity is resolved.